libs/api/ebpf_api.cpp

```diff
@@ -67,10 +67,9 @@ ebpf_api_create_map(
 {
     UNREFERENCED_PARAMETER(map_flags);
 
-    _ebpf_operation_create_map_request request{
-        sizeof(_ebpf_operation_create_map_request),
-        ebpf_operation_id_t::EBPF_OPERATION_CREATE_MAP,
-        {sizeof(struct _ebpf_map_definition), type, key_size, value_size, max_entries}};
+    _ebpf_operation_create_map_request request{sizeof(_ebpf_operation_create_map_request),
+                                               ebpf_operation_id_t::EBPF_OPERATION_CREATE_MAP,
+                                               {sizeof(struct bpf_map), type, key_size, value_size, max_entries}};
```

Comment:
bpf_map is #def to _ebpf_map_definition in ebpf_helper.h which will be only included by ebpf programs. So I think we cannot use "struct bpf_map" instead of "struct _ebpf_map_definition" here?

Also bpf_map will defined as _ebpf_map in this file's context.

Reply:
No that's unrelated.  The use here is a different definition of bpf_map which is defined in api_internal.h.   (There are two separate bpf_map structs in standard bpf... one in the execution context and one used in libbpf.)

Reply:
bpf_map in api_internal is defined as follows:

`typedef struct bpf_map
{
    const struct bpf_object* object;
    char* name;
    ebpf_handle_t map_handle;
    fd_t map_fd;
    bool pinned;
} ebpf_map_t;`

This is different than `struct _ebpf_map_definition` that we are replacing here?

Reply:
reverted